src/PubSub/Drivers/LocalClient.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace BeyondCode\LaravelWebSockets\PubSub\Drivers;
+
+use stdClass;
+use React\EventLoop\LoopInterface;
+use React\Promise\FulfilledPromise;
+use React\Promise\PromiseInterface;
+use BeyondCode\LaravelWebSockets\PubSub\ReplicationInterface;
+
+class LocalClient implements ReplicationInterface
+{
+    /**
+     * Mapping of the presence JSON data for users in each channel.
+     *
+     * @var string[][]
+     */
+    protected $channelData = [];
+
+    /**
+     * Boot the pub/sub provider (open connections, initial subscriptions, etc).
+     *
+     * @param LoopInterface $loop
+     * @return self
+     */
+    public function boot(LoopInterface $loop) : ReplicationInterface
+    {
+        return $this;
+    }
+
+    /**
+     * Publish a payload on a specific channel, for a specific app.
+     *
+     * @param string $appId
+     * @param string $channel
+     * @param stdClass $payload
+     * @return bool
+     */
+    public function publish(string $appId, string $channel, stdClass $payload) : bool
+    {
+        // Nothing to do, nobody to publish to
+        return true;
+    }
+
+    /**
+     * Subscribe to receive messages for a channel.
+     *
+     * @param string $appId
+     * @param string $channel
+     * @return bool
+     */
+    public function subscribe(string $appId, string $channel) : bool
+    {
+        return true;
+    }
+
+    /**
+     * Unsubscribe from a channel.
+     *
+     * @param string $appId
+     * @param string $channel
+     * @return bool
+     */
+    public function unsubscribe(string $appId, string $channel) : bool
+    {
+        return true;
+    }
+
+    /**
+     * Add a member to a channel. To be called when they have
+     * subscribed to the channel.
+     *
+     * @param string $appId
+     * @param string $channel
+     * @param string $socketId
+     * @param string $data
+     */
+    public function joinChannel(string $appId, string $channel, string $socketId, string $data)
+    {
+        $this->channelData["$appId:$channel"][$socketId] = $data;
```

Comment:
It's done this way because the channel subscription key (at least in the Redis client) is a single string made up of those two parts. I did it that way in LocalClient to mirror that. I'm not against changing it if you have a good reason to, though.

Reply:
Hmm well besides proper practice. It's useful to easily (& fast) get all channels from a certain appID. There's no benefit of doing it like "$appId:$channel" imho apart from the fact that redis uses x:y & to mirror that :)